malcolm/core/serializable.py

```diff
@@ -200,6 +206,12 @@ def lookup_subclass(cls, d):
         Returns:
             Serializable subclass
         """
-        typeid = d["typeid"]
-        subclass = cls._subcls_lookup[typeid]
-        return subclass
+        try:
+            typeid = d["typeid"]
+        except KeyError:
+            raise KeyError("typeid field not present in JSON message")
+        try:
+            subclass = cls._subcls_lookup[typeid]
+            return subclass
+        except KeyError as e:
+            raise KeyError('%s not a valid malcolm message subclass' % str(e))
```

Comment:
maybe do cls._subcls_lookup.get(typeid, None)

if subclass is None:
    raise KeyError('%s not a valid typeid" % typeid)

is not a valid typeid